st2common/st2common/query/base.py

```diff
@@ -126,7 +126,9 @@ def _query_and_save_results(self, query_context, last_query_time=None):
                 LOG.debug('Removed state object %s.', query_context)
             return
 
-        if status in action_constants.LIVEACTION_COMPLETED_STATES:
+        if (status in action_constants.LIVEACTION_COMPLETED_STATES or
+                status == action_constants.LIVEACTION_STATUS_PAUSED):
```

Comment:
The invoke post run is needed here. In short, this will tell a super workflow that this task that is a st2 action of another mistral workflow has paused.